datafusion/core/src/execution/context.rs

```diff
@@ -492,6 +497,15 @@ impl SessionContext {
         }
     }
 
+    /// Creates a [`DataFrame`] that will execute the specified
+    /// LogicalPlan, but will error if the plans represent DDL such as
+    /// `CREATE TABLE`
+    ///
+    /// Use [`Self::dataframe`] to run plans with DDL
+    pub fn dataframe_without_ddl(&self, plan: LogicalPlan) -> Result<DataFrame> {
```

Comment:
This is the API to support https://github.com/apache/arrow-datafusion/issues/4720

Reply:
This appears to just be DataFrame new, do we really need this?